pyboy/plugins/game_wrapper_pokemon_red/wrapper.py

```diff
@@ -0,0 +1,385 @@
+#
+# License: See LICENSE.md file
+# GitHub: https://github.com/Baekalfen/PyBoy
+#
+
+__pdoc__ = {
+    "GameWrapperPokemonRed.cartridge_title": False,
+    "GameWrapperPokemonRed.post_tick": False,
+}
+
+from pyboy.utils import WindowEvent
+from pyboy.logger import logger
+from ..base_plugin import PyBoyGameWrapper
+from . import constants
+from .utils import get_character_index
+
+PKMN_SIZE = 0x2C
+BYTE_ORDER = 'big'
+
+
+try:
+    from cython import compiled
+    cythonmode = compiled
+except ImportError:
+    cythonmode = False
+
+
+class GameWrapperPokemonRed(PyBoyGameWrapper):
+    """
+    This class wraps Pokemon Red, and provides easy access to score and a "fitness" score for AIs.
+
+    If you call `print` on an instance of this object, it will show an overview of everything this object provides.
+    """
+    cartridge_title = "POKEMON RED"
+
+    def __init__(self, *args, **kwargs):
+        # TODO: Determine proper size
+        self.shape = (20, 16)
+        """The shape of the game area"""
+        super().__init__(*args, game_area_section=(0, 0) + self.shape, game_area_wrap_around=True, **kwargs)
+
+    def start_game(self, timer_div=None):
+        """
+        Call this function right after initializing PyBoy. This will navigate through menus to start the game at the
+        first playable state.
+
+        The state of the emulator is saved, and using `reset_game`, you can get back to this point of the game
+        instantly.
+
+        Kwargs:
+            timer_div (int): Replace timer's DIV register with this value. Use `None` to randomize.
+        """
+        while self.get_player_name() != 'NINTEN': # Player is NINTEN on start screen
+            self.pyboy.tick()
+        self.pyboy.send_input(WindowEvent.PRESS_BUTTON_START)
+        self.pyboy.tick()
+        PyBoyGameWrapper.start_game(self, timer_div=timer_div)
+
+    def reset_game(self, timer_div=None):
+        """
+        After calling `start_game`, you can call this method at any time to reset the game.
+
+        Kwargs:
+            timer_div (int): Replace timer's DIV register with this value. Use `None` to randomize.
+        """
+        PyBoyGameWrapper.reset_game(self, timer_div=timer_div)
+
+    def get_int_at_address(self, address, size=2):
+        """Returns an integer from a given address of a specified size."""
+        bytes = []
+        for i in range(size):
+            bytes.append(self.pyboy.get_memory_value(address + i))
+        return int.from_bytes(bytes, byteorder=BYTE_ORDER)
+
+    def set_int_at_address(self, address, value, size=2):
+        """Sets an integer at a given address of a specified size."""
+        bytes = value.to_bytes(2, byteorder=BYTE_ORDER)
+        i = 0
+        for byte in bytes:
+            self.pyboy.set_memory_value(address + i, byte)
+            i += 1
```

Comment:
These probably belong elsewhere. Maybe on the PyBoy object itself. Are these for the user of the plugin?